### ISSUE TITLE
Current status on 22.05 kHz voices

### ISSUE THREAD

**Issue by entenbein** [marytts/marytts#441]
Hi folks,

I started trying to build a 22050Hz voice and got an error at the F0CARTTrainer stage, [similar to this one](http://www.dfki.de/pipermail/mary-users/2011-July/000918.html). So this probably also derives from the 22kHz input.

What is the current status of this. Is it supported at all? What would I need to change except from the `db.samplingrate` and the audio files of course?

Thanks

**Comment by entenbein** [comment]
Anyone?

**Comment by psibre** [comment]
This is a *really* old enhancement request going back several years. I was sure we had an old ticket or two open, but can't find it atm.

If you would like to tackle this issue, feel free. I would suggest trying to build a 22.05 kHz voice and see (and fix) step-by-step where it fails to work as expected.

**TL;DR:** there are several points scattered throughout the code where explicitly (or implicitly) the assumption of 16 kHz sampling rate is hard-coded. In several cases this is an upstream issue because HTS for many years needed 16 kHz audio data. In other cases, it's really just ~~sloppy programming~~ a lack of QA.

**Comment by entenbein** [comment]
Alright, I'll let you know how it goes!

**Comment by entenbein** [comment]
Well guess what, I started running all stages again, waiting for the error described above to appear at the F0CARTTrainer stage ... But everything worked.

The only difference was that this time I built a HTK instead of EHMM voice.
